build.gradle

```diff
@@ -535,6 +541,59 @@ configure(subprojects - getJsProjects()) { subproject ->
     }
 }
 
+def dexerProjects = getAndroidProjects().findAll {
+    !it.name.contains("-specs-")
+}
+
+configure(dexerProjects) { subproject ->
+    afterEvaluate {
+        def checkDexer = task('checkDexer', type: JavaExec, description: 'compiles android project jar file into dex bytecode', dependsOn: jar) {
+            doFirst {
+                mkdir "${subproject.buildDir}/kotlin/d8"
+            }
```

Comment:
it just came to my mind that gradle supports this via outputs
```
outputs.dir("${subproject.buildDir}/kotlin/d8")
```

Reply:
will try updating